nixos/modules/programs/wayland/uwsm.nix

```diff
@@ -0,0 +1,122 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.uwsm;
+
+  # Helper function to create desktop entry files for UWSM-managed compositors
+  mk_uwsm_desktop_entry =
+    opts:
+    (pkgs.writeTextFile {
+      name = "${opts.name}_uwsm";
+      text = ''
+        [Desktop Entry]
+        Name=${opts.compositorPrettyName} (UWSM)
+        Comment=${opts.compositorComment}
+        Exec=${lib.getExe cfg.package} start -S -F "${opts.compositorBinPath}"
+        Type=Application
+      '';
+      destination = "/share/wayland-sessions/${opts.name}-uwsm.desktop";
+      derivationArgs = {
+        passthru.providedSessions = [ "${opts.name}-uwsm" ];
+      };
+    });
+in
+{
+  options.programs.uwsm = {
+    enable = lib.mkEnableOption ''
+      uwsm, which wraps standalone Wayland compositors with a set
+      of Systemd units on the fly. This essentially
+      binds the wayland compositor into `graphical-session-pre.target`,
+      `graphical-session.target`, `xdg-desktop-autostart.target`.
+
+      This is useful for Wayland compositors like Hyprland, Sway, Wayfire,
+      etc. that do not start these targets and services on their own.
+
+      If you are having trouble starting a service that depends on
+      `graphical-session.target`, while using a WM, enabling this option
+      might help.
+
+      Note: you must configure `waylandCompositors` suboptions as well
```

Comment:
This should render prettier. [Source](https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#admonitions)
```suggestion
      ::: {.note}
      You must configure `waylandCompositors` suboptions as well
      :::
```

Reply:
Sweet.

![image](https://github.com/user-attachments/assets/a3d94f0b-4615-4a62-ac5d-2750162fc9eb)